### DISCUSSION TITLE
Support higher dimension vectors

### DISCUSSION THREAD

**Discussion by julianlumets** [supabase/supabase#29193] [Feature Requests]
Hi team!

I wanted to create a RAG app using embeddings with a dimension size of 3072. However i followed this tutorial (https://github.com/supabase-community/chatgpt-your-files) and at the step:

create index on document_sections using hnsw (embedding vector_ip_ops);

Supabase SQL Editor returns the error:

"not more than 2000 dimensions allowed"

Could you increase the dimension size to support newer embedding models?

Thank you!

**Comment by ngoldbla** [comment]
I am trying to use the latest openai large v3 embeddings but supabase vector db cannot handle that many dimensions

**Comment by zaporter** [comment]
This is a pgvector limitation, not supabase. See: https://github.com/pgvector/pgvector/issues/461 and https://github.com/pgvector/pgvector#hnsw

I solved this by switching to halfvec

**Reply by ngoldbla** [reply]
Hey @zaporter thanks so much for solving this and sharing the answer!

**Reply by lucasbojakowski** [reply]
hey, how did you achieve this switch? i'm going over the deps code and changing stuff but i cant seem to make it work, and i can't find a easy way to use halfvec
